test/plugins/utils/messageCarrier.js

```diff
@@ -0,0 +1,158 @@
+const expect = require('expect');
+const MessageCarrier = require('../../../src/plugins/utils/messageCarrier');
+
+describe('MessageCarrier tests', function() {
+  it('get errors returns all errors added', function() {
+    const messages = new MessageCarrier();
+
+    messages.addMessage(['paths', '/example', 'get'], 'message1', 'error');
+    messages.addMessage(['paths', '/example', 'post'], 'message2', 'error');
+
+    expect(messages.errors.length).toEqual(2);
+    expect(messages.errors[0]).toEqual({
+      path: ['paths', '/example', 'get'],
+      message: 'message1'
+    });
+    expect(messages.errors[1]).toEqual({
+      path: ['paths', '/example', 'post'],
+      message: 'message2'
+    });
+    expect(messages.warnings.length).toEqual(0);
```

Comment:
I appreciate the completeness here but since this is a unit test for the `get errors` method, you shouldn't add a dependency on the `get warnings` method. This applies to the next test as well